pkg/ifaces/watcher.go

```diff
@@ -45,27 +47,40 @@ func NewWatcher(bufLen int) *Watcher {
 func (w *Watcher) Subscribe(ctx context.Context) (<-chan Event, error) {
 	out := make(chan Event, w.bufLen)
 
-	nsHandles, err := getNetNSHandles()
+	netns, err := getNetNS()
 	if err != nil {
-		go w.sendUpdates(ctx, netns.None(), out)
+		go w.sendUpdates(ctx, "", out)
 	} else {
-		for _, nsh := range nsHandles {
-			nsHandle := nsh
-			go w.sendUpdates(ctx, nsHandle, out)
+		for _, n := range netns {
+			go w.sendUpdates(ctx, n, out)
```

Comment:
You're removing the capturing of `n` which was done on purpose .. is it intentional? (I trying to understand if this issue is fixed in go1.22, cf https://go.dev/blog/loopvar-preview , but the blog pre-dates the 1.22 release so I'm not sure)

Reply:
yes I dropped it after moving to go1.22 as I believe this behavior if fixed now